ui/src/pages/version/components/forms/components/ResourcesPanel.js

```diff
@@ -69,12 +69,14 @@ export const ResourcesPanel = ({
       setGpuValueOptions([{ value: "None", inputDisplay: "None" }]);
     }
 
-    if (resourcesConfig.min_replica > maxAllowedReplica) {
-      onChange("min_replica")(maxAllowedReplica - 1)
-    }
+    if (!!maxAllowedReplica) {
```

Comment:
We only trigger the changes to min/max replicas when an environment is selected (a separate asynchronous effect sets the `maxAllowedReplica` to a non-undefined value when this happens).

Reply:
`non-undefined`, you mean the setEffect sets a value when env is selected?

Reply:
Yep, it sets it [here](https://github.com/caraml-dev/merlin/blob/0a79211db6ed60b266ee75cf03088b4f22eb037d/ui/src/pages/version/components/forms/components/DeploymentConfigPanel.js#L93).